plugin-opentelemetry/src/output.rs

```diff
@@ -0,0 +1,156 @@
+use alumet::{
+    measurement::{MeasurementBuffer, WrappedMeasurementValue},
+    pipeline::elements::{error::WriteError, output::OutputContext},
+};
+use anyhow::Context;
+use opentelemetry::{global, InstrumentationScope, KeyValue};
+use opentelemetry_otlp::{MetricExporter, WithExportConfig};
+use opentelemetry_sdk::metrics::SdkMeterProvider;
+use opentelemetry_sdk::Resource;
+use std::{env, sync::OnceLock};
+
+#[derive(Clone)]
+pub struct OpenTelemetryOutput {
+    append_unit_to_metric_name: bool,
+    use_unit_display_name: bool,
+    add_attributes_to_labels: bool,
+    prefix: String,
+    suffix: String,
+    initialized: bool,
+    collector_host: String,
+}
+
+impl OpenTelemetryOutput {
+    pub fn new(
+        append_unit_to_metric_name: bool,
+        use_unit_display_name: bool,
+        add_attributes_to_labels: bool,
+        prefix: String,
+        suffix: String,
+        collector_host: String,
+    ) -> anyhow::Result<OpenTelemetryOutput> {
+        Ok(Self {
+            append_unit_to_metric_name,
+            use_unit_display_name,
+            add_attributes_to_labels,
+            prefix,
+            suffix,
+            initialized: false,
+            collector_host: format!("{}{}", collector_host, "/v1/metrics"),
+        })
+    }
+    pub fn initialize(&mut self) {
+        // Needs to be created inside the tokio thread
```

Comment:
Can you add a TODO so that we remember to change that once #119 is fixed? Ideally, this would be done in the builder, but here it's not possible because output builders lack the runtime context.